### PR TITLE
Deprecate download from LGTM by hiding it behind canary flag

### PR DESCRIPTION
**Pull request by koesie10**
This will hide the download from LGTM functionality behind the canary config setting, effectively removing it for all normal users.

## Checklist

- [x] [CHANGELOG.md](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.
- [ ] Issues have been created for any UI or other user-facing changes made by this pull request.
- [ ] _[Maintainers only]_ If this pull request makes user-facing changes that require documentation changes, open a corresponding docs pull request in the [github/codeql](https://github.com/github/codeql/tree/main/docs/codeql/codeql-for-visual-studio-code) repo and add the `ready-for-doc-review` label there.

### CODE REVIEW

File: extensions/ql-vscode/package.json [hunk 1]
```diff
@@ -930,6 +930,10 @@
           "command": "codeQL.chooseDatabaseGithub",
           "when": "config.codeQL.canary"
         },
+        {
```
[the remark this answers is not in the document]
Comment (koesie10) [reply extensions/ql-vscode/package.json:933]: Yes, this adds a condition for the command to be shown in the command palette. The default is for the command to be shown (so essentially a `when` of `true`), so this adds a condition to showing the command in the command palette.

**Review by charisk — APPROVED** [review]
LGTM!

**Comment by robertbrignull** [comment]
PR looks good but a note we don't want to merge this just yet. Need to get a few other things done first.

**Comment by robertbrignull** [comment]
Should be good to merge now. We have the documentation and changelog post ready.

**Pull request merged**